openupgrade_scripts/scripts/account/15.0.1.2/pre-migration.py

```diff
@@ -0,0 +1,409 @@
+                    WHEN refund_tax.type_tax_use = 'purchase'
+                        OR (invoice_tax.type_tax_use = 'sale'
+                            AND atpl.refund_tax_id IS NULL)
+                    THEN TRUE
+                    ELSE FALSE
+                END AS tax_tag_invert
+            FROM account_move am
+            JOIN account_move_line aml ON am.id = aml.move_id
+            JOIN account_tax_repartition_line atpl ON aml.tax_repartition_line_id = atpl.id
+            LEFT JOIN account_tax refund_tax ON refund_tax.id = atpl.refund_tax_id
+            LEFT JOIN account_tax invoice_tax ON invoice_tax.id = atpl.invoice_tax_id
+            WHERE am.move_type = 'entry'
+        ) sub
+        WHERE sub.id = aml.id AND aml.tax_tag_invert IS NULL
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move_line aml
+        SET tax_tag_invert = sub.tax_tag_invert
+        FROM (
+            SELECT DISTINCT ON (aml.id) aml.id,
+                CASE
+                    WHEN (tax.type_tax_use = 'purchase' AND (
+                            aml.credit != 0 AND aml.credit IS NOT NULL))
+                        OR (tax.type_tax_use = 'sale' AND (
+                            aml.debit = 0 OR aml.debit IS NULL))
+                    THEN TRUE
+                    ELSE FALSE
+                END AS tax_tag_invert
+            FROM account_move am
+            JOIN account_move_line aml ON am.id = aml.move_id
+            JOIN account_move_line_account_tax_rel tax_rel ON
+                aml.id = tax_rel.account_move_line_id
+            JOIN account_tax tax ON tax.id = tax_rel.account_tax_id
+            WHERE am.move_type = 'entry'
+        ) sub
+        WHERE sub.id = aml.id AND aml.tax_tag_invert IS NULL""",
+    )
+
+
+def _create_account_payment_method_line(env):
+    # Create account_payment_method_line table
+    sql.create_model_table(
+        env.cr,
+        "account_payment_method_line",
+        columns=[
+            ("name", "varchar", ""),
+            ("sequence", "integer", ""),
+            ("payment_method_id", "integer", ""),
+            ("payment_account_id", "integer", ""),
+            ("journal_id", "integer", ""),
+            ("create_uid", "integer", ""),
+            ("create_date", "timestamp", ""),
+            ("write_uid", "integer", ""),
+            ("write_date", "timestamp", ""),
+        ],
+    )
+    # Create account payment method lines from account payment methods
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO account_payment_method_line (name, sequence,
+            payment_method_id, journal_id, create_uid, write_uid,
+            create_date, write_date)
+        SELECT apm.name, 10, apm.id, aj.id,
+            apm.create_uid, apm.write_uid, apm.create_date, apm.write_date
+        FROM account_payment_method apm, account_journal aj
+        WHERE apm.code = 'manual' AND aj.type IN ('bank', 'cash')
```

Comment:
Did you consider to precreate the payment method lines for check_printing and the payment acquiers here as well?

Reply:
No, I just did them in https://github.com/OCA/OpenUpgrade/pull/3513 and https://github.com/OCA/OpenUpgrade/pull/3515.